### ISSUE TITLE
unable to log in

### ISSUE THREAD

**Issue by kolko123** [alik0211/mtproto-core#40]
I use this code: https://github.com/alik0211/mtproto-core/blob/master/docs/authentication.md

I enter my api_id and api_hash and number, but writes auth.authorization: undefined

**Comment by somgupta13** [comment]
`const { MTProto } = require('@mtproto/core');
const prompt = require('prompt');

const mtproto = new MTProto({
api_id: 105,
api_hash: '85b728',

test: false,
});

const phone = '+91808';
const code = '2222';
// const password = 'hellojs';

function sendCode(phone) {
console.log("hello");
return mtproto.call('auth.sendCode', {
api_id: 1056,
api_hash: '85b72864f343b734',
	// current_number: false,
phone_number: phone,
settings: {
_: 'codeSettings',
},
})
// .catch(error=>{console.log(error)});
}

sendCode(phone)
.catch(error => {
if (error.error_message.includes('_MIGRATE_')) {
      const [type, nextDcId] = error.error_message.split('_MIGRATE_');

      mtproto.setDefaultDc(+nextDcId);

      return sendCode(phone).catch(error=>{console.log(error)});;
}
})
.then( result => {
f1(result.phone_code_hash);

})

async function f1(phone_code_hash){
prompt.start();
await prompt.get(['username'], async function (err, result) {
if (err) { console.log(err); }
else{
// console.log(phone,phone_code_hash,result.username);
// console.log(typeof(result.username));

 		const { user } = await mtproto.call('auth.signIn', {
			phone_number   : phone,
			phone_code_hash: phone_code_hash,
			phone_code     : result.username
		}).then(result => {
			console.log("Logged:", result);
			console.log("Logged:", result.user);
			 
return result;
}).catch(error => {
			console.log("ff")
console.log(error);
process.exit();
});
	
 }
});
}

`

**Comment by kolko123** [comment]
how should i get the code?

**Comment by kolko123** [comment]
> `const { MTProto } = require('@mtproto/core');
> const prompt = require('prompt');
> 
> const mtproto = new MTProto({
> api_id: 105,
> api_hash: '85b728',
> 
> test: false,
> });
> 
> const phone = '+91808';
> const code = '2222';
> // const password = 'hellojs';
> 
> function sendCode(phone) {
> console.log("hello");
> return mtproto.call('auth.sendCode', {
> api_id: 1056,
> api_hash: '85b72864f343b734',
> // current_number: false,
> phone_number: phone,
> settings: {
> _: 'codeSettings',
> },
> })
> // .catch(error=>{console.log(error)});
> }
> 
> sendCode(phone)
> .catch(error => {
> if (error.error_message.includes('_MIGRATE_')) {
> const [type, nextDcId] = error.error_message.split('_MIGRATE_');
> 
> ```
>   mtproto.setDefaultDc(+nextDcId);
> 
>   return sendCode(phone).catch(error=>{console.log(error)});;
> ```
> 
> }
> })
> .then( result => {
> f1(result.phone_code_hash);
> 
> })
> 
> async function f1(phone_code_hash){
> prompt.start();
> await prompt.get(['username'], async function (err, result) {
> if (err) { console.log(err); }
> else{
> // console.log(phone,phone_code_hash,result.username);
> // console.log(typeof(result.username));
> 
> ```
> 	const { user } = await mtproto.call('auth.signIn', {
> 		phone_number   : phone,
> 		phone_code_hash: phone_code_hash,
> 		phone_code     : result.username
> 	}).then(result => {
> 		console.log("Logged:", result);
> 		console.log("Logged:", result.user);
> ```
> 
> return result;
> }).catch(error => {
> console.log("ff")
> console.log(error);
> process.exit();
> });
> 
> }
> });
> }
> 
> `

hello
(node:17837) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'phone_code_hash' of undefined

**Comment by somgupta13** [comment]
https://github.com/somgupta13/Telegram-API/blob/master/som.js

**Comment by kolko123** [comment]
> https://github.com/somgupta13/Telegram-API/blob/master/som.js

same problem

**Comment by somgupta13** [comment]
Do you have node 12version?

**Comment by kolko123** [comment]
> Do you have node 12version?

v12.16.2

**Comment by somgupta13** [comment]
code is working for me

**Comment by kolko123** [comment]
> code is working for me

I should specify only api_id and api_hash and phone number?

**Comment by somgupta13** [comment]
yes

**Comment by kolko123** [comment]
> yes

а что-то еще должно быть установлено?

**Comment by somgupta13** [comment]
are you recieving otp?

**Comment by kolko123** [comment]
> are you recieving otp?

what is it?

**Comment by somgupta13** [comment]
are you recieving the otp on telegram number

**Comment by kolko123** [comment]
i don't get anything

> are you recieving the otp on telegram number

**Comment by somgupta13** [comment]
type your mobile number in international format

**Comment by kolko123** [comment]
> type your mobile number in international format

it is introduced in an international format

**Comment by somgupta13** [comment]
https://github.com/somgupta13/Telegram-API/blob/master/som.js
try this i have updated else show your code

**Comment by kolko123** [comment]
> https://github.com/somgupta13/Telegram-API/blob/master/som.js
> try this i have updated else show your code

same problem

**Comment by alik0211** [comment]
@kolko123 
1. What version of the library do you use?
2. Show your code
3. Is Telegram blocked in your country? VPN enabled (try open [telegram.org](https://telegram.org/))?

**Comment by alik0211** [comment]
@kolko123 if you don't send an info by may 11 I'll close the issue

**Comment by flaviolacer** [comment]
Version 3.1.0 keep stuck on promise

ex:

```node
client.call('auth.signIn', {
            phone_number: state.phone_number,
            phone_code_hash: state.phone_code_hash,
            phone_code: state.code
        }).then(message => {
            console.log("Logged:", message); // never happen
            return message;  // never happen
        }).catch(error => {
            console.log(error);
            process.exit();
        });
```

On telegram app says that i logged.

**Comment by alik0211** [comment]
This error is probably fixed in version **3.1.1**. View more: https://t.me/mtproto_core/17
